web/test/pages/roadmap.test.tsx

```diff
@@ -228,4 +239,132 @@ describe("roadmap page", () => {
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+  });
+
+  it("expands start section only", () => {
+    useMockRoadmap({
+      steps: [
+        generateStep({
+          name: "step1",
+          section: "PLAN",
+          tasks: [generateTask({ id: "task1" })],
+        }),
+        generateStep({
+          name: "step2",
+          section: "START",
+          tasks: [generateTask({ id: "task2" })],
+        }),
+      ],
+    });
+
+    useMockUserData({
+      preferences: generatePreferences({
+        roadmapOpenSections: ["START"],
+      }),
+    });
+
+    const subject = renderRoadmapPage();
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).not.toBeVisible();
+  });
+
+  it("expands plan section only", () => {
+    useMockRoadmap({
+      steps: [
+        generateStep({
+          name: "step1",
+          section: "PLAN",
+          tasks: [generateTask({ id: "task1" })],
+        }),
+        generateStep({
+          name: "step2",
+          section: "START",
+          tasks: [generateTask({ id: "task2" })],
+        }),
+      ],
+    });
+
+    useMockUserData({
+      preferences: generatePreferences({
+        roadmapOpenSections: ["PLAN"],
+      }),
+    });
+
+    const subject = renderRoadmapPage();
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).not.toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+  });
+
+  it("updates accordion state", async () => {
+    useMockRoadmap({
+      steps: [
+        generateStep({
+          name: "step1",
+          section: "PLAN",
+          tasks: [generateTask({ id: "task1" })],
+        }),
+        generateStep({
+          name: "step2",
+          section: "START",
+          tasks: [generateTask({ id: "task2" })],
+        }),
+      ],
+    });
+
+    const userData = generateUserData({
+      preferences: generatePreferences({
+        roadmapOpenSections: ["PLAN", "START"],
+      }),
+    });
+
+    setupStatefulUserDataContext();
+    const subject = render(
+      <WithStatefulUserData initialUserData={userData}>
+        <RoadmapPage displayContent={{ contentMd: "" }} />)
+      </WithStatefulUserData>
+    );
+
+    const sectionPlan = subject.container.querySelector("#plan-header");
+    expect(sectionPlan).toBeInTheDocument();
+    fireEvent.click(sectionPlan as Element);
+    expect(currentUserData().preferences.roadmapOpenSections).toEqual(["START"]);
```

Comment:
it seems like we're testing only half of that "if (expanded) else if (!expanded)" block in the handleAccordionStateChange function. We're only testing the removal of something from the roadmapOpenSections array. I could delete the entire contents of your "else if" block right now and the tests would still pass.

Reply:
As an aside, this is one of the reasons that I personally like to practice TDD. When you write the failing tests first and only write enough code to get them passing, then you can ensure that your tests cover all the possible paths of your code branches.

Reply:
Addressed.